File: daspro/hapus_arr.cpp
#include <stdio.h>

int main() {
    // List nama-nama buah
    char buah[4][10] = {"apel", "anggur", "mangga", "Jeruk"};
    int jumlahBuah = 4; // Jumlah elemen dalam array
    int i, j;

    // Menghapus elemen di indeks ke-2 (mangga)
    for (i = 2; i < jumlahBuah - 1; i++) {
        for (j = 0; buah[i + 1][j] != '\0'; j++) {
            buah[i][j] = buah[i + 1][j];
        }
        buah[i][j] = '\0'; // Menambahkan karakter null di akhir string
    }
    jumlahBuah--; // Update jumlah buah

    // Tampilkan semua daftar buah
    printf("Semua Buah: ada %d macam\n", jumlahBuah);
    for (i = 0; i < jumlahBuah; i++) {
        printf("%s\n", buah[i]);
    }

    return 0;
}

File: daspro/hapus_arr.pas
program HapusArray;
uses crt;

var
    buah: array[0..3] of string[10]; // Array dengan panjang 4 elemen
    jumlahBuah, i, j, indexToDelete: Integer;

begin
    clrscr;

    // Inisialisasi array dengan beberapa buah
    buah[0] := 'apel';
    buah[1] := 'anggur';
    buah[2] := 'mangga';
    buah[3] := 'jeruk';

    jumlahBuah := 4; // Jumlah buah awal
    indexToDelete := 2; // Indeks yang ingin dihapus (indeks ke-2 adalah "mangga")

    // Menghapus elemen di indeks ke-2 (mangga)
    for i := indexToDelete to jumlahBuah - 2 do
        for j := 1 to Length(buah[i+1]) do
            buah[i][j] := buah[i+1][j]; // Geser karakter satu per satu
    jumlahBuah := jumlahBuah - 1; // Update jumlah buah

    // Tampilkan semua daftar buah
    Writeln('Semua Buah: ada', jumlahBuah, 'macam');
    for i := 0 to jumlahBuah - 1 do
        Writeln(buah[i]);

    readln;
end.
